### PR TITLE
Updated vscode syntax to cover most of the language

### PR DESCRIPTION
**Pull request by dbaileychess**
Updated the syntax for vscode to include most of the language features.

There is still a bunch to do (`generics`, `fn`, `specializers` etc) but this PR covers most things well.

Added a new example file that tries to cover all the grammar cases, so that it is easy to see if the syntax highlighting is working over the whole language.

### CODE REVIEW

File: samples/syntax_highlight.lobster [hunk 1]
```diff
@@ -0,0 +1,134 @@
+// This is an example Lobster file to demonstrate syntax highlighting.
```
Comment (aardappel) [inline samples/syntax_highlight.lobster:1]: can we move this file to `tests` ?

**Review by aardappel — COMMENTED** [review]
Nice.. thanks for being so thorough :)

File: samples/syntax_highlight.lobster (hunk 1 above)
Comment (dbaileychess) [reply samples/syntax_highlight.lobster]: Done!

**Comment by dbaileychess** [comment]
I thought it was a good way to learn the language, at least the grammar.

**Pull request merged**